nodejs/src/index.js

```diff
@@ -22,9 +26,18 @@ exports.write = function (struct, compression, callback) {
 };
 
 exports.read = function (Model, rawData, callback) {
+	var callback = arguments.length === 4 ? arguments[3] : arguments[2];
```

Comment:
Why not just call the last of the arguments. `arguments.pop()` (which is nastily mutatey) or `arguments[arguments.length-1]` ?